benches/centralized_telescope.rs

```diff
@@ -0,0 +1,201 @@
+//! Benchmarking centralized_telescope
+
+use criterion::{black_box, criterion_group, criterion_main, measurement::WallTime, Criterion};
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+use std::time::{Duration, Instant};
+
+use alba::centralized_telescope::{algorithm, init, params::Params, setup::Setup};
+
+pub mod criterion_helpers;
+use criterion_helpers::{benchmarks, Repetitions, Steps};
+
+#[path = "../src/utils/test_utils.rs"]
+mod test_utils;
+
+// Global variables
+const NAME: &str = "Centralized";
+// Benchmark parameters
+const L: &[f64] = &[50.0, 128.0]; // Security parameter
+const SP: &[u64] = &[1_000]; // Size of set to lower bound
+const NP: &[u64] = &[80, 90, 95, 98]; // Alba's np parameter, |Sp| >= np
+const NF: &[u64] = &[67, 75]; // Alba's nf parameter,  |Sp| >= np > nf
+
+/// Function generating a random set of elements to bench and calling Alba's centralized setup
+pub fn centralized_setup(
+    rng: &mut ChaCha20Rng,
+    l: f64,
+    sp: u64,
+    np: u64,
+    nf: u64,
+) -> (Vec<[u8; 32]>, Setup) {
+    let seed_u32 = rng.next_u32();
+    let seed = seed_u32.to_ne_bytes().to_vec();
+    let dataset: Vec<[u8; 32]> = test_utils::gen_items(&seed, sp);
+    let params = Params {
+        soundness_param: l,
+        completeness_param: l,
+        set_size: np.saturating_mul(sp).div_ceil(100),
+        lower_bound: nf.saturating_mul(sp).div_ceil(100),
+    };
+    (dataset, init::make_setup(&params))
+}
+
+/// Bench the duration of both the proving and verifiying algorithm of Alba centralized
+fn time_benches(c: &mut Criterion) {
+    #[allow(clippy::unit_arg)]
+    fn prove_duration(l: f64, sp: u64, np: u64, nf: u64, truncate_size: u64, n: u64) -> Duration {
+        let mut rng = ChaCha20Rng::from_entropy();
+        let mut total_duration = Duration::ZERO;
+        for _ in 0..n {
+            // Setup
+            let (mut dataset, bench_setup) = centralized_setup(&mut rng, l, sp, np, nf);
+            // Truncate the dataset to give truncate_size elements to the prover
+            dataset.truncate(truncate_size as usize);
+            // Bench
+            let start = Instant::now();
+            black_box(algorithm::prove(&bench_setup, &dataset));
+            total_duration = total_duration.saturating_add(start.elapsed());
+        }
+        total_duration
+    }
+
+    #[allow(clippy::unit_arg)]
+    fn verify_duration(l: f64, sp: u64, np: u64, nf: u64, truncate_size: u64, n: u64) -> Duration {
```

Comment:
Yeah verify should at most run some hundreds of hashes so is quite fast. It may still be good to display a precise number, especially when dealing with highly time sensitive apps.